scripts/download-realm.js

```diff
@@ -156,43 +137,100 @@ function getSyncCommitSha(version) {
+            desired.CORE_ARCHIVE = `realm-core-${flavor}-v${dependencies.REALM_CORE_VERSION}-Darwin-devel.tar.gz`;
+            return Promise.resolve();
+        case 'ios':
+            flavor = flavor === 'Debug' ? 'MinSizeDebug' : flavor;
+            desired.CORE_SERVER_FOLDER += `/ios/${flavor}`;
+            desired.CORE_ARCHIVE = `realm-core-${flavor}-v${dependencies.REALM_CORE_VERSION}-iphoneos.tar.gz`;
+            return Promise.resolve();
+        case 'win':
+            if (!options.arch) throw new Error(`Specifying '--arch' is required for platform 'win'`);
+            const arch = options.arch === 'ia32' ? 'Win32' : options.arch;
+            desired.CORE_SERVER_FOLDER += `/windows/${arch}/nouwp/${flavor}`;
+            desired.CORE_ARCHIVE = `realm-core-${flavor}-v${dependencies.REALM_CORE_VERSION}-Windows-${arch}-devel.tar.gz`;
+            return Promise.resolve();
+        case 'linux':
+            desired.CORE_SERVER_FOLDER = 'core';
+            desired.CORE_ARCHIVE = `realm-core-${dependencies.REALM_CORE_VERSION}.tgz`;
+            desired.CORE_ARCHIVE_ROOT = `realm-core-${dependencies.REALM_CORE_VERSION}`;
+            return Promise.resolve();
+        default:
+            return Promise.reject(new Error(`Unsupported sync platform '${options.platform}'`));
+    }
+}
+
+function appendSyncRequirements(desired, dependencies, options) {
+    desired.SYNC_SERVER_FOLDER = 'sync';
     let flavor = options.debug ? 'Debug' : 'Release';
 
-    let archive, archiveRootFolder;
-    let promise = Promise.resolve();
     switch (options.platform) {
-    case 'mac':
-        archive = `realm-sync-node-cocoa-${dependencies.REALM_SYNC_VERSION}.tar.gz`;
-        archiveRootFolder = `realm-sync-node-cocoa-${dependencies.REALM_SYNC_VERSION}`;
-        break;
-    case 'ios':
-        archive = `realm-sync-cocoa-${dependencies.REALM_SYNC_VERSION}.tar.xz`;
-        archiveRootFolder = `core`;
-        break;
-    case 'win':
-        promise = acquireCore(options, dependencies, target)
-                  .then(() => getSyncCommitSha(dependencies.REALM_SYNC_VERSION))
-                  .then(sha => serverFolder += `/sha-version/${sha}`);
-        const arch = options.arch === 'ia32' ? 'Win32' : options.arch;
-        archive = `realm-sync-${flavor}-v${dependencies.REALM_SYNC_VERSION}-Windows-${arch}-devel.tar.gz`;
-        break;
-    default:
-        throw new Error(`Unsupported sync platform '${options.platform}'`);
+        case 'mac':
+            desired.SYNC_ARCHIVE = `realm-sync-node-cocoa-${dependencies.REALM_SYNC_VERSION}.tar.gz`;
+            desired.SYNC_ARCHIVE_ROOT = `realm-sync-node-cocoa-${dependencies.REALM_SYNC_VERSION}`;
+            return Promise.resolve();
+        case 'ios':
+            desired.SYNC_ARCHIVE = `realm-sync-cocoa-${dependencies.REALM_SYNC_VERSION}.tar.xz`;
+            desired.SYNC_ARCHIVE_ROOT = `core`;
+            return Promise.resolve();
```

Comment:
It seems the old code did this but why not resolve with the `desired` object and then skip the additional `.then( () => desired)`. below

Reply:
at line 197